From 2d68aa84711cfbc2d052def90cb20f4d07301aa8 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 4 Apr 2019 16:56:57 +0800
Subject: [PATCH 06/30] drm/imx/dcss: use drm_gem_fb_prepare_fb() instead of
 drm_fb_cma_prepare_fb

In upstream commit c0f095f76698 ("drm/fb-cma-helper: Remove unused functions"),
drm_fb_cma_prepare_fb is substituted for drm_gem_fb_prepare_fb, and
drm_fb_cma_create has been substituted for drm_gem_fb_create.

Because of upstream commit 29ad20b22c8f ("drm: Add drm_device->fb_helper pointer"),
drm_device holds a pointer to the drm_fb_helper structure. This means that the driver
doesn't have to keep track of that. Also use the drm_fb_helper functions directly.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-kms.c   | 13 ++++---------
 drivers/gpu/drm/imx/dcss/dcss-plane.c |  3 ++-
 drivers/gpu/drm/imx/dpu/dpu-kms.c     | 13 ++++---------
 drivers/gpu/drm/imx/dpu/dpu-plane.c   |  3 ++-
 4 files changed, 12 insertions(+), 20 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-kms.c b/drivers/gpu/drm/imx/dcss/dcss-kms.c
index 5fc7ea7ef5f2..076a3cb2da4d 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-kms.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-kms.c
@@ -20,17 +20,12 @@
 #include <drm/drm_gem_cma_helper.h>
 #include <linux/dma-buf.h>
 #include <linux/reservation.h>
+#include <drm/drm_gem_framebuffer_helper.h>
+#include <drm/drm_fb_helper.h>
 
 #include "imx-drm.h"
 #include "dcss-crtc.h"
 
-static void dcss_drm_output_poll_changed(struct drm_device *drm)
-{
-	struct imx_drm_device *imxdrm = drm->dev_private;
-
-	drm_fbdev_cma_hotplug_event(imxdrm->fbhelper);
-}
-
 static int dcss_drm_atomic_check(struct drm_device *drm,
 				 struct drm_atomic_state *state)
 {
@@ -201,8 +196,8 @@ static int dcss_drm_atomic_commit(struct drm_device *dev,
 }
 
 const struct drm_mode_config_funcs dcss_drm_mode_config_funcs = {
-	.fb_create = drm_fb_cma_create,
-	.output_poll_changed = dcss_drm_output_poll_changed,
+	.fb_create = drm_gem_fb_create,
+	.output_poll_changed = drm_fb_helper_output_poll_changed,
 	.atomic_check = dcss_drm_atomic_check,
 	.atomic_commit = dcss_drm_atomic_commit,
 };
diff --git a/drivers/gpu/drm/imx/dcss/dcss-plane.c b/drivers/gpu/drm/imx/dcss/dcss-plane.c
index 0c75f6dd2e9e..e2854d6ad6de 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-plane.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-plane.c
@@ -19,6 +19,7 @@
 #include <drm/drm_gem_cma_helper.h>
 #include <drm/drm_atomic.h>
 #include <linux/dma-buf.h>
+#include <drm/drm_gem_framebuffer_helper.h>
 
 #include "video/imx-dcss.h"
 #include "video/viv-metadata.h"
@@ -630,7 +631,7 @@ static void dcss_plane_atomic_disable(struct drm_plane *plane,
 }
 
 static const struct drm_plane_helper_funcs dcss_plane_helper_funcs = {
-	.prepare_fb = drm_fb_cma_prepare_fb,
+	.prepare_fb = drm_gem_fb_prepare_fb,
 	.atomic_check = dcss_plane_atomic_check,
 	.atomic_update = dcss_plane_atomic_update,
 	.atomic_disable = dcss_plane_atomic_disable,
diff --git a/drivers/gpu/drm/imx/dpu/dpu-kms.c b/drivers/gpu/drm/imx/dpu/dpu-kms.c
index 9371904fdf17..0c674dc8445a 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-kms.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-kms.c
@@ -13,10 +13,12 @@
  */
 
 #include <drm/drmP.h>
+#include <drm/drm_fb_helper.h>
 #include <drm/drm_atomic.h>
 #include <drm/drm_atomic_helper.h>
 #include <drm/drm_fb_cma_helper.h>
 #include <drm/drm_gem_cma_helper.h>
+#include <drm/drm_gem_framebuffer_helper.h>
 #include <linux/dma-buf.h>
 #include <linux/reservation.h>
 #include <linux/sort.h>
@@ -25,13 +27,6 @@
 #include "dpu-plane.h"
 #include "imx-drm.h"
 
-static void dpu_drm_output_poll_changed(struct drm_device *dev)
-{
-	struct imx_drm_device *imxdrm = dev->dev_private;
-
-	drm_fbdev_cma_hotplug_event(imxdrm->fbhelper);
-}
-
 static struct drm_plane_state **
 dpu_atomic_alloc_tmp_planes_per_crtc(struct drm_device *dev)
 {
@@ -773,8 +768,8 @@ static int dpu_drm_atomic_check(struct drm_device *dev,
 }
 
 const struct drm_mode_config_funcs dpu_drm_mode_config_funcs = {
-	.fb_create = drm_fb_cma_create,
-	.output_poll_changed = dpu_drm_output_poll_changed,
+	.fb_create = drm_gem_fb_create,
+	.output_poll_changed = drm_fb_helper_output_poll_changed,
 	.atomic_check = dpu_drm_atomic_check,
 	.atomic_commit = drm_atomic_helper_commit,
 };
diff --git a/drivers/gpu/drm/imx/dpu/dpu-plane.c b/drivers/gpu/drm/imx/dpu/dpu-plane.c
index ca49176a7624..f29fbc09245f 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-plane.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-plane.c
@@ -20,6 +20,7 @@
 #include <drm/drm_plane_helper.h>
 #include <video/dpu.h>
 #include <video/imx8-prefetch.h>
+#include <drm/drm_gem_framebuffer_helper.h>
 #include "dpu-plane.h"
 #include "imx-drm.h"
 
@@ -908,7 +909,7 @@ static void dpu_plane_atomic_update(struct drm_plane *plane,
 }
 
 static const struct drm_plane_helper_funcs dpu_plane_helper_funcs = {
-	.prepare_fb = drm_fb_cma_prepare_fb,
+	.prepare_fb = drm_gem_fb_prepare_fb,
 	.atomic_check = dpu_plane_atomic_check,
 	.atomic_update = dpu_plane_atomic_update,
 };
-- 
2.17.1

